container/container.go

```diff
@@ -914,6 +916,19 @@ func (container *Container) BuildCreateEndpointOptions(n libnetwork.Network, epC
 		}
 	}
 
+	var dns []string
+
+	if len(container.HostConfig.DNS) > 0 {
+		dns = container.HostConfig.DNS
+	} else if len(epConfig.DNS) > 0 {
+		dns = epConfig.DNS
```

Comment:
Why is that ? Please take a look at `buildSandboxOptions` and it is exactly the same logic I prefer here.
It will be great if we can not replicate the same code in 2 places. Infact this is the reason for my comment in libnetwork to avoid adding the new endpoint specific option for DNS. But since Sandbox is created after endpoint create, we didnt have a choice. Anyways, that is not a big deal. we can use the same logic as buildSandboxOptions and that should work. 

```
        if len(container.HostConfig.DNS) > 0 {
                dns = container.HostConfig.DNS
        } else if len(daemon.configStore.DNS) > 0 {
                dns = daemon.configStore.DNS
        }

        for _, d := range dns {
                sboxOptions = append(sboxOptions, libnetwork.OptionDNS(d))
        }
```